tests/test_merge_db_functions.py

```diff
@@ -0,0 +1,337 @@
+              'ConvOclDirectFwdGen,0.0536,0,miopenConvolutionFwdAlgoDirect,not used',
+          'miopenConvolutionFwdAlgoGEMM':
+              'GemmFwdRest,0.05712,2749200,miopenConvolutionFwdAlgoGEMM,not used'
+      },
+      '1-19-19-1x1-64-19-19-1024-0x0-1x1-1x1-0-NCHW-FP32-B': {
+          'miopenConvolutionBwdDataAlgoWinograd':
+              'ConvBinWinogradRxSf2x3g1,0.186719,0,miopenConvolutionBwdDataAlgoWinograd,not used',
+          'miopenConvolutionBwdDataAlgoDirect':
+              'ConvOclDirectFwd1x1,0.234239,0,miopenConvolutionBwdDataAlgoDirect,not used',
+          'miopenConvolutionBwdDataAlgoGEMM':
+              'GemmBwd1x1_stride1,0.239519,0,miopenConvolutionBwdDataAlgoGEMM,not used',
+          'miopenConvolutionBwdDataAlgoImplicitGEMM':
+              'ConvAsmImplicitGemmGTCDynamicBwdXdlopsNHWC,0.327199,94633984,miopenConvolutionBwdDataAlgoImplicitGEMM,not used'
+      }
+  }
+
+  key = '1-160-698-5x5-64-79-348-1-1x1-2x2-1x1-0-NCHW-FP32-F'
+  vals = {
+      'miopenConvolutionFwdAlgoImplicitGEMM':
+          'ConvMlirIgemmFwdXdlops,0.03776,0,miopenConvolutionFwdAlgoImplicitGEMM,not used'
+  }
+  keep_keys = False
+  target_merge(master_list, key, vals, keep_keys)
+
+
+def test_update_master_list():
+
+  master_list = {
+      '1-48-480-3x3-16-48-480-1-1x1-1x1-1x1-0-NCHW-FP16-F': {
+          'miopenConvolutionFwdAlgoImplicitGEMM':
+              'ConvHipImplicitGemmForwardV4R4Xdlops_Padded_Gemm,0.02512,0,miopenConvolutionFwdAlgoImplicitGEMM,not used',
+          'miopenConvolutionFwdAlgoDirect':
+              'ConvOclDirectFwd,0.02544,0,miopenConvolutionFwdAlgoDirect,not used',
+          'miopenConvolutionFwdAlgoWinograd':
+              'ConvBinWinogradRxSf2x3g1,0.03104,0,miopenConvolutionFwdAlgoWinograd,not used',
+          'miopenConvolutionFwdAlgoGEMM':
+              'GemmFwdRest,0.06096,414720,miopenConvolutionFwdAlgoGEMM,not used'
+      }
+  }
+
+  key = '1-160-698-5x5-64-79-348-1-1x1-2x2-1x1-0-NCHW-FP32-F'
+  local_paths = ['/data/tests/../utils/test_files/usr_gfx90a68.HIP.fdb.txt']
+  mids = [-1]
+  keep_keys = False
+  update_master_list(master_list, local_paths, mids, keep_keys)
+
+
+def test_write_merge_results():
+
+  master_list = {
+      '1-48-480-3x3-16-48-480-1-1x1-1x1-1x1-0-NCHW-FP16-F': {
+          'miopenConvolutionFwdAlgoImplicitGEMM':
+              'ConvHipImplicitGemmForwardV4R4Xdlops_Padded_Gemm,0.02512,0,miopenConvolutionFwdAlgoImplicitGEMM,not used',
+          'miopenConvolutionFwdAlgoDirect':
+              'ConvOclDirectFwd,0.02544,0,miopenConvolutionFwdAlgoDirect,not used',
+          'miopenConvolutionFwdAlgoWinograd':
+              'ConvBinWinogradRxSf2x3g1,0.03104,0,miopenConvolutionFwdAlgoWinograd,not used',
+          'miopenConvolutionFwdAlgoGEMM':
+              'GemmFwdRest,0.06096,414720,miopenConvolutionFwdAlgoGEMM,not used'
+      }
+  }
+
+  final_file = '/data/tests/old_gfx90a68.HIP.fdb.txt'
+  copy_files = []
+
+  write_merge_results(master_list, final_file, copy_files)
```

Comment:
this is also missing an assert. We need to check for expected values

Reply:
Added the the assert to cehck the final file result file content.